Source/WebKit/UIProcess/Launcher/cocoa/ExtensionProcess.mm

```diff
@@ -84,7 +88,7 @@
     return grant;
 }
 
-PlatformGrant ExtensionProcess::grantCapability(const PlatformCapability& capability) const
+PlatformGrant ExtensionProcess::grantCapability(const PlatformCapability& capability, Function<void()>&& invalidationHandler) const
```

Comment:
Could this take an `AssertionCapability` instead of a `PlatformCapability` as an argument and just use the `m_didInvalidateBlock` from `AssertionCapability`? It seems a bit weird to be passing the invalidation block in two different places...

Reply:
That's a good point! I will make that change.

Reply:
I started on this change, but it became more involving than expected, because this method is also used to grant Media capabilities. Would you be OK with the current approach, Ben?